quill-jdbc/src/main/scala/io/getquill/sources/jdbc/ConfigJdbcSource.scala

```diff
@@ -0,0 +1,17 @@
+package io.getquill.sources.jdbc
+
+import java.sql.Connection
+
+import io.getquill.JdbcSourceConfig
+import io.getquill.naming.NamingStrategy
+import io.getquill.sources.sql.idiom.SqlIdiom
+
+import scala.util.DynamicVariable
+
+class ConfigJdbcSource[D <: SqlIdiom, N <: NamingStrategy](config: JdbcSourceConfig[D, N])
+  extends JdbcSource[D, N](config) {
+
+  private val currentConnection = new DynamicVariable[Option[Connection]](Some(dataSource.getConnection))
```

Comment:
the `DynamicVariable` shouldn't be necessary anymore, since the transaction will be propagated explicitly instead of through a `ThreadLocal`